setup.py

```diff
@@ -69,6 +69,42 @@
     ],
     package_data = {'nltk': ['test/*.doctest', 'VERSION']},
     install_requires = ['six>=1.10.0'],
+    extras_require = {
+        'all': [
+            'nose>=1.3.0',
+            'tox>=1.6.1',
+            'coverage>=3.7.1',
+            'pylint>=1.1.0',
+            'numpy>=1.8.0',
+            'scipy>=0.13.2',
+            'matplotlib>=1.3.1',
+            'scikit-learn>=0.14.1',
+            'python-crfsuite>=0.8.2',
+            'gensim>=0.11.1',
+            'pyparsing>=2.0.3',
+            'twython>=3.2.0'
+        ],
+        'machine_learning': [
+            'gensim>=0.11.1',
+            'numpy>=1.8.0',
+            'pyparsing>=2.0.3',
```

Comment:
I can see an import in [`tgrep.py`](https://github.com/nltk/nltk/blob/develop/nltk/tgrep.py#L118).